app/models/goods_nomenclature.rb

```diff
@@ -17,15 +17,19 @@ class GoodsNomenclature
   has_many :ancestors, polymorphic: true
 
   def validity_start_date=(validity_start_date)
-    @attributes['validity_start_date'] = Date.parse(validity_start_date.to_s) if validity_start_date.present?
+    return if validity_start_date.blank?
+
+    @attributes['validity_start_date'] = validity_start_date.to_date
```

Comment:
### Why did I make this change?
- `to_s` is deprecated
-  `to_fs` does not work with String, and validity_start_date sometimes is a String (and sometimes a TimeZone)
- `date_to` works with string and other Date compatible classes.

Reply:
Just thinking this one through - it works great as is and replicates the existing behaviour _but_ it does highlight its not possible to change the value to nil

Our use case is read only so I don't see us having a model with a date assigned, then changing it back to nil so this isn't an issue here but for future reference if it was 

```ruby
@attributes['validity_start_date'] = validity_start_date.presence&.to_date
```

then it would handle blanks but also allow assigning to `nil`